// @llamaindex/liteparse/dist/lib.js
import { native, } from "./native.js";
// ---------------------------------------------------------------------------
// LiteParse class
// ---------------------------------------------------------------------------
export class LiteParse {
    _native;
    _config;
    constructor(userConfig = {}) {
        const nativeConfig = {
            ocrLanguage: userConfig.ocrLanguage,
            ocrEnabled: userConfig.ocrEnabled,
            ocrServerUrl: userConfig.ocrServerUrl,
            tessdataPath: userConfig.tessdataPath,
            maxPages: userConfig.maxPages,
            targetPages: userConfig.targetPages,
            dpi: userConfig.dpi,
            outputFormat: userConfig.outputFormat,
            preserveVerySmallText: userConfig.preserveVerySmallText,
            password: userConfig.password,
            quiet: userConfig.quiet,
            numWorkers: userConfig.numWorkers,
        };
        this._native = new native.LiteParse(nativeConfig);
        // Read back the resolved config from the native side
        const resolved = this._native.config;
        this._config = {
            ocrLanguage: resolved.ocrLanguage ?? "eng",
            ocrEnabled: resolved.ocrEnabled ?? true,
            ocrServerUrl: resolved.ocrServerUrl ?? undefined,
            tessdataPath: resolved.tessdataPath ?? undefined,
            maxPages: resolved.maxPages ?? 1000,
            targetPages: resolved.targetPages ?? undefined,
            dpi: resolved.dpi ?? 150,
            outputFormat: resolved.outputFormat ?? "json",
            preserveVerySmallText: resolved.preserveVerySmallText ?? false,
            password: resolved.password ?? undefined,
            quiet: resolved.quiet ?? false,
            numWorkers: resolved.numWorkers ?? 1,
        };
    }
    async parse(input) {
        // Convert Uint8Array to Buffer for the native side
        const nativeInput = typeof input === "string" ? input : Buffer.from(input);
        const result = await this._native.parse(nativeInput);
        return {
            pages: result.pages.map(toPage),
            text: result.text,
        };
    }
    async screenshot(input, pageNumbers) {
        const nativeInput = typeof input === "string" ? input : Buffer.from(input);
        const results = await this._native.screenshot(nativeInput, pageNumbers ?? null);
        return results.map((r) => ({
            pageNum: r.pageNum,
            width: r.width,
            height: r.height,
            imageBuffer: r.imageBuffer,
        }));
    }
    getConfig() {
        return { ...this._config };
    }
}
function toPage(p) {
    return {
        pageNum: p.pageNum,
        width: p.width,
        height: p.height,
        text: p.text,
        textItems: p.textItems.map(toTextItem),
    };
}
function toTextItem(item) {
    return {
        text: item.text,
        x: item.x,
        y: item.y,
        width: item.width,
        height: item.height,
        fontName: item.fontName,
        fontSize: item.fontSize,
        confidence: item.confidence,
    };
}
export function searchItems(items, options) {
    const nativeResults = native.searchItems(items, options.phrase, options.caseSensitive ?? false);
    return nativeResults.map(toTextItem);
}
export default LiteParse;
